### DISCUSSION TITLE
Postinstall script not executing after publish

### DISCUSSION THREAD

**Discussion by un-focused** [npm/feedback#762] [General]
None of my scripts are working after publishing an npm package. I will try to give a simple example here to demonstrate the point.

The package.json scripts object (relevant parts) looks like this:

```
"scripts": {
  "install": "node scripts/install.js",
  "postinstall": "npx prisma generate"
}
```
The scripts/install.js looks like this:

```
function main() {
    console.log('hello world')
    process.exit(0);
}

main();
```

This works when I execute:

```
$ npm I
```

inside the package that I am publishing.

When I install it locally it works too:

```
$ npm i
$ npm run build
$ npm pack --pack-destination
```

After the pack, in another project, I specify the path to the tar file generated & everything works!

If I run:

```
$ npm publish
```

& then install it inside the same project that consumes the node_module, it does not execute the install or postinstall script.

To debug this, I run the command:

```
npm i --foreground-scripts
```

and have tried npm i --foreground-scripts --loglevel verbose as well to better understand the problem.

The documentation for the npm lifecycle is lacking & I could not find any information online. Please note that this package is publish in a private gitlab registry and I am able to download & "install" the package as I see it in my node_modules but it does run the lifecycle scripts when installing remotely from npm.

I do have an .npmrc file specified.

Not sure if this is a bug so thought I would open it to discussion first!

This is my first post & I hope I have been descriptive & clear. Any help would be appreciated :).

**Comment by un-focused** [comment]
This is the same issue on both my Mac machines: iMac 2019 & M1 Pro MBPro running npm 8.8.0

**Comment by un-focused** [comment]
I tested this with npm 7 & the issue is the same but seems be more compatible with the packages!

**Comment by ljharb** [comment]
The install script simply won’t work the way you expect - avoid defining it. Postinstall works, but output from postinstall scripts is suppressed.

**Reply by un-focused** [reply]
Yes, which is why I used:

```
npm i --foreground-scripts
```

to see the output. Still no output from my package when installing remotely.

**Reply by un-focused** [reply]
To clarify, there is output from other packages

**Reply by ljharb** [reply]
right - "install" isn't a script that works the way you expect. Put it all in postinstall.

**Reply by un-focused** [reply]
Yes, I did that too. I tried it in both places. To clarify, this works when installing I npm pack the package & point to the tar file.

**Reply by un-focused** [reply]
```
"postinstall": "node scripts/postinstall.js"
```

I renamed install.js to postinstall.js

**Comment by un-focused** [comment]
This works on npm 6.14.2. It looks like this issue started in npm 7 & persists in npm 8.
